### PR TITLE
832 dzelge loading sea ice data failed

### PR DESCRIPTION
**Pull request by dzelge**
When accepting this pull request cate with trow a ValidationError when loading invalid lat/lon
coordinates from a data source when applying a region constraint. 

Lat/lons are invalid if:

- lat or lon does not exist
- lat or lon has the the wrong dimension (!=1)

### CODE REVIEW

**Comment by dzelge** [comment]
> 
> 
> Let's fix the builds first.

Done

File: CHANGES.md [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 ## Version 2.0.0.dev24
 
+* Loading SeaIce data throws a ValueError: The truth value of an array with more than one element is ambiguous.
```
Comment (forman) [inline CHANGES.md]:
```suggestion
* Fixed failing download of Sea-Ice CCI data  (ValueError: The truth value of an array with more than one element is ambiguous)
```

File: cate/core/op.py [hunk 2]
```diff
@@ -698,12 +698,12 @@ def new_subprocess_op(op_meta_info: OpMetaInfo,
            and returns a tuple (label, total_work) or a regex that must match
            in order to signal the start of progress monitoring.
            The regex must provide the group names "label" or "total_work" or both,
-           e.g. "(?P<label>\w+)" or "(?P<total_work>\d+)"
+           e.g. "(?P<label>\\w+)" or "(?P<total_work>\\d+)"
```
Comment (forman) [inline cate/core/op.py]: Revert to master

File: cate/core/op.py [hunk 3]
```diff
@@ -698,12 +698,12 @@ def new_subprocess_op(op_meta_info: OpMetaInfo,
...
     :param progress: Either a callable that receives a text line from the executable's stdout
            and returns a tuple (work, msg) or a regex that must match
            in order to signal process.
            The regex must provide group names "work" or "msg" or both,
-           e.g. "(?P<msg>\w+)" or "(?P<work>\d+)"
+           e.g. "(?P<msg>\\w+)" or "(?P<work>\\d+)"
```
Comment (forman) [inline cate/core/op.py]: Revert to master

File: cate/core/opimpl.py [hunk 4]
```diff
@@ -805,6 +809,16 @@ def subset_spatial_impl(ds: xr.Dataset,
...
     """
+
+    # Validate whether lat and lon exists.
+
+    if not hasattr(ds, 'lon') or not hasattr(ds, 'lat'):
+        raise ValidationError('Cannot apply regional subset. No (valid) geocoding found.')
+
+    if hasattr(ds, 'lon') and len(ds.lon.shape) != 1 \
+            or hasattr(ds, 'lat') and len(ds.lat.shape) != 1:
+        raise ValidationError('Geocoding not recognised. Lat and/or lon variables have more than one dimension.')
```
Comment (forman) [inline cate/core/opimpl.py]:
```suggestion
        raise ValidationError('Geocoding not recognised. Variables "lat" and/or "lon" have more than one dimension.')
```

File: cate/core/workflow.py [hunk 5]
```diff
@@ -1109,11 +1109,11 @@ class SubProcessStep(OpStepBase):
     :param started_re: A regex that must match a text line from the process' stdout
            in order to signal the start of progress monitoring.
            The regex must provide the group names "label" or "total_work" or both,
-           e.g. "(?P<label>\w+)" or "(?P<total_work>\d+)"
+           e.g. "(?P<label>\\w+)" or "(?P<total_work>\\d+)"
```
Comment (forman) [inline cate/core/workflow.py]: Revert to master

File: cate/core/workflow.py [hunk 6]
```diff
@@ -1109,11 +1109,11 @@ class SubProcessStep(OpStepBase):
...
     :param progress_re: A regex that must match a text line from the process' stdout
            in order to signal process.
            The regex must provide group names "work" or "msg" or both,
-           e.g. "(?P<msg>\w+)" or "(?P<work>\d+)"
+           e.g. "(?P<msg>\\w+)" or "(?P<work>\\d+)"
```
Comment (forman) [inline cate/core/workflow.py]: Revert to master

File: cate/ops/animate.py [hunk 7]
```diff
@@ -47,6 +47,7 @@
 
 import os
 
+# import matplotlib
```
Comment (forman) [inline cate/ops/animate.py]: Revert to master

File: cate/util/misc.py [hunk 8]
```diff
@@ -411,7 +411,7 @@ def new_indexed_name(names: Iterable[str], pattern: str) -> str:
     """
     if "{index}" not in pattern:
         raise ValueError('pattern must contain "{index}"')
-    re_pattern = re.compile(pattern.replace("{index}", "(\d+)"))
+    re_pattern = re.compile(pattern.replace("{index}", r"(\d+)"))
```
Comment (forman) [inline cate/util/misc.py]: Is that tested?

File: cate/util/process.py [hunk 9]
```diff
@@ -148,11 +148,11 @@ class ProcessOutputMonitor:
     :param started: Either a callable that receives a text line and returns a tuple (label, total_work)
            or a regex that must match in order to signal the start of progress monitoring.
            The regex must provide the group names "label" or "total_work" or both,
-           e.g. "(?P<label>\w+)" or "(?P<total_work>\d+)"
+           e.g. r"(?P<label>\\w+)" or "(?P<total_work>\\d+)"
```
Comment (forman) [inline cate/util/process.py]: Revert to master

File: cate/util/process.py [hunk 10]
```diff
@@ -148,11 +148,11 @@ class ProcessOutputMonitor:
...
     :param progress: Either a callable that receives a text line and returns a tuple (work, msg)
            or a regex that must match in order to signal process.
            The regex must provide group names "work" or "msg" or both,
-           e.g. "(?P<msg>\w+)" or "(?P<work>\d+)"
+           e.g. r"(?P<msg>\\w+)" or "(?P<work>\\d+)"
```
Comment (forman) [inline cate/util/process.py]: Revert to master

File: cate/util/process.py (hunk 9 above)
Comment (forman) [inline cate/util/process.py]: Revert to master

File: cate/util/web/webapi.py [hunk 11]
```diff
@@ -699,7 +699,7 @@ def url_pattern(pattern: str):
     :return: equivalent regex pattern
     :raise ValueError: if *pattern* is invalid
     """
-    name_pattern = '(?P<%s>[^\\;\\/\\?\\:\\@\\&\\=\\+\\$\\,]+)'
+    name_pattern = r'(?P<%s>[^\;\/\?\:\@\&\=\+\$\,]+)'
```
Comment (forman) [inline cate/util/web/webapi.py]: Revert to master

**Pull request merged**